### PR TITLE
Add error log to BLInterlockCtrl device

### PR DESCRIPTION
**Pull request by xresende**
So as to have most diagnostics for failures while opening the beamline.

### CODE REVIEW

**Review by fernandohds564 — APPROVED** [review]

File: siriuspy/siriuspy/devices/intlkctrl.py [hunk 1]
```diff
@@ -149,86 +149,136 @@ def __init__(self, devname=None, props2init='all', **kwargs):
...
+        """Return error log for last check method invoked."""
+        return self._error_log
+
     @property
     def is_hutchA_intlk_search_done(self):
         """."""
-        return self['A:PPS01:SEARCH_OK'] == 1
+        status_ok = self['A:PPS01:SEARCH_OK'] == 1
+        if not status_ok:
+            self._error_log = 'hutch A interlock search not Ok.'
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/intlkctrl.py:165]: I think error log should return to empty string if command is successful

**Review by anacso17 — CHANGES_REQUESTED** [review]

File: siriuspy/siriuspy/devices/intlkctrl.py (hunk 1 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/intlkctrl.py:165]: Contrary to what I talked to you offline moments agor, ana, I thought about it and I think now that it is more useful for `error_log` to record not the current error state of the BL but the last error observed during the process of opening the BL, regardless of when/what point this happened. For this we should not return empty string in `is_*` methods when they succeed but only when `cmd_beamline_open` specifically succeeds.

Comment (xresende) [reply siriuspy/siriuspy/devices/intlkctrl.py]: yes. done.

File: siriuspy/siriuspy/devices/intlkctrl.py [hunk 2]
```diff
@@ -268,68 +297,68 @@ def is_hutchB_gatevalves_closed(self):
...
+                self.is_machine_gamma_enabled and \
+                self.is_beamline_eps_ok and \
+                self.is_frontend_shutter_eps_permission_ok and \
+                self.is_hutchA_shutter_eps_permission_ok and \
+                self.is_frontend_gamma_shutter_opened and \
+                self.is_frontend_photon_shutter_opened and \
+                self.is_hutchA_gamma_shutter_opened:
+            return True
+        else:
             return False
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/intlkctrl.py:361]: you could return the if condition here

File: siriuspy/siriuspy/devices/intlkctrl.py [hunk 3]
```diff
@@ -453,60 +481,73 @@ def cmd_hutchA_photon_close(self, timeout=None):
...
                 if _time.time() - t0 > self.TIMEOUT_GATEVALVE:
-                    msg = 'open frontend and hutchA gatevalve timeout reached!'
-                    print(msg)
+                    print(self.error_log)
                     return False
                 _time.sleep(0.5)
 
         # check hutchA shutter permission and open gatevalves for hutchB
         if not self.is_hutchA_shutter_eps_permission_ok:
-            print('open hutchB gatevalves')
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/intlkctrl.py:493]: this removal was on purpose?

File: siriuspy/siriuspy/devices/intlkctrl.py (hunk 2 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/intlkctrl.py:361]: ah, ok. It is True. or False.
done. thanks!

File: siriuspy/siriuspy/devices/intlkctrl.py (hunk 3 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/intlkctrl.py:493]: maybe your comment is deprecated ?

**Review by anacso17 — APPROVED** [review]

File: siriuspy/siriuspy/devices/intlkctrl.py (hunk 3 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/intlkctrl.py:493]: yes, I am printing now only if command, which is executed if necessary, fails.

**Pull request merged**